label_studio/ml/api.py

```diff
@@ -227,29 +283,65 @@ def post(self, request, *args, **kwargs):
+        return generics.get_object_or_404(Task, pk=validated_data['task'], project=ml_backend.project)
+
+    def _get_credentials(self, request, context, project):
+        if flag_set('ff_back_dev_2362_project_credentials_060722_short', request.user):
+            context.update(
+                project_credentials_login=project.task_data_login,
+                project_credentials_password=project.task_data_password,
+            )
+        return context
+
+    def _get_ml_results(self, ml_api_result):
+        results = ml_api_result.response.get('results', [None])
+        if isinstance(results, list) and len(results) >= 1:
+            return results[0]
+
+        return None
+
     def post(self, request, *args, **kwargs):
+        """
+        Send a request to the machine learning backend set up to be used for interactive preannotations to retrieve a
+        predicted region based on annotator input.
+        """
         ml_backend = generics.get_object_or_404(MLBackend, pk=self.kwargs['pk'])
         self.check_object_permissions(self.request, ml_backend)
         serializer = MLInteractiveAnnotatingRequest(data=request.data)
         serializer.is_valid(raise_exception=True)
-        validated_data = serializer.validated_data
 
-        task = generics.get_object_or_404(Task, pk=validated_data['task'], project=ml_backend.project)
-        context = validated_data.get('context')
+        task = self._get_task(ml_backend, serializer.validated_data)
+        context = self._get_credentials(request, serializer.validated_data.get('context', {}), task.project)
 
-        if flag_set('ff_back_dev_2362_project_credentials_060722_short', request.user):
-            context['project_credentials_login'] = task.project.task_data_login
-            context['project_credentials_password'] = task.project.task_data_password
+        ml_api_result = ml_backend.interactive_annotating(task, context, user=self.request.user)
+
+        if ml_api_result.is_error:
+            message = f'Prediction not created for project {self}: {ml_api_result.error_message}'
+            return self._error_response(message)
+
+        if not isinstance(ml_api_result.response, dict) or 'results' not in ml_api_result.response:
+            message = f'Incorrect response from ML service it must be a dict and contain "results" key: {ml_api_result.response}'
+            return self._error_response(message)
+
+        ml_results = self._get_ml_results(ml_api_result)
 
-        result = ml_backend.interactive_annotating(task, context, user=request.user)
+        if not ml_results:
+            message = f'ML backend has to return a list with at least 1 annotation but it returned: {type(ml_results)}'
+            return self._error_response(message, logger.warning)
 
-        return Response(
-            result,
-            status=status.HTTP_200_OK,
-        )
+        return Response({'data': ml_results}, status=status.HTTP_200_OK)
```

Comment:
There are three test failures coming from this code, because `ml_api_result` at the top is a dictionary, not an `MLApiResult`, and therefore doesn't have any methods like `is_error`. The tests that need to be fixed are:

```
tests/test_predictions.py::test_interactive_annotating
tests/test_predictions.py::test_interactive_annotating_failing
tests/test_predictions.py::test_interactive_annotating_with_drafts
```